### PR TITLE
Various fixes to ensure system-probe tests cleanup properly

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Fixes a bunch of small issues in tests resulting in incomplete cleanup.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Review by xornivore — APPROVED** [review]

**Review by xornivore — APPROVED** [review]

**Review by xornivore — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

File: pkg/network/protocols/kafka/server.go [hunk 1]
```diff
@@ -21,5 +22,5 @@ func RunServer(t testing.TB, serverAddr, serverPort string) error {
 
 	t.Helper()
 	dir, _ := testutil.CurDir()
-	return protocolsUtils.RunDockerServer(t, "kafka", dir+"/testdata/docker-compose.yml", env, regexp.MustCompile(".*started \\(kafka.server.KafkaServer\\).*"), protocolsUtils.DefaultTimeout)
+	return protocolsUtils.RunDockerServer(t, "kafka", dir+"/testdata/docker-compose.yml", env, regexp.MustCompile(".*started \\(kafka.server.KafkaServer\\).*"), 2*time.Minute)
```
Comment (guyarb) [inline pkg/network/protocols/kafka/server.go:25]: Can you explain why did you change this?

File: pkg/network/protocols/testutil/serverutils.go [hunk 2]
```diff
@@ -85,8 +88,8 @@ func RunHostServer(t *testing.T, command []string, env []string, serverStartRege
 	err := cmd.Start()
 	require.NoErrorf(t, err, "could not start %s on host", serverName)
 	t.Cleanup(func() {
-		_ = cmd.Process.Kill()
-		_ = cmd.Process.Release()
+		cancel()
+		_ = cmd.Wait()
```
Comment (guyarb) [inline pkg/network/protocols/testutil/serverutils.go:92]: Does it have a timeout? otherwise it might "stuck" us

**Review by guyarb — CHANGES_REQUESTED** [review]
Left a single comment

File: pkg/network/protocols/kafka/server.go (hunk 1 above)
Comment (brycekahle) [reply pkg/network/protocols/kafka/server.go:25]: I was getting timeouts in CI

**Review by guyarb — APPROVED** [review]
